parcels/field.py

```diff
@@ -222,9 +227,15 @@ def __init__(self, name, data, lon=None, lat=None, depth=None, time=None, grid=N
             if len(self.data.shape) == 4:
                 self.data = self.data.reshape(sum(((self.data.shape[0],), self.data.shape[2:]), ()))
         if len(self.data.shape) == 4:
-            assert self.data.shape == (self.grid.tdim, self.grid.zdim, self.grid.ydim, self.grid.xdim)
+            assert self.data.shape == (self.grid.tdim, self.grid.zdim, self.grid.ydim, self.grid.xdim), \
+                                      ('Field %s expect a data shape of a [tdim, zdim, ydim, xdim] = [%d %d %d %d], but data is [%d %d %d %d]' %
+                                       (self.name, self.grid.tdim, self.grid.zdim, self.grid.ydim, self.grid.xdim,
+                                        self.data.shape[0], self.data.shape[1], self.data.shape[2], self.data.shape[3]))
         else:
-            assert self.data.shape == (self.grid.tdim, self.grid.ydim, self.grid.xdim)
+            assert self.data.shape == (self.grid.tdim, self.grid.ydim, self.grid.xdim), \
+                                      ('Field %s expect a data shape of a [tdim, ydim, xdim] = [%d %d %d], but data is [%d %d %d]' %
```

Comment:
change `Field %sexpect` to `Field %s: expecting`

Also, add that flag `transpose=True` might be used to fix this